lib/connection_pool.rb

```diff
@@ -53,21 +53,14 @@ def initialize(options = {}, &block)
   end
 
   def with(options = {})
-    # Connections can become corrupted via Timeout::Error.  Discard
-    # any connection whose usage after checkout does not finish as expected.
-    # See #67
-    success = false
-    conn = checkout(options)
-    begin
-      result = yield conn
-      success = true # means the connection wasn't interrupted
-      result
-    ensure
-      if success
-        # everything is roses, we can safely check the connection back in
+    Thread.handle_interrupt(StandardError => :never) do
```

Comment:
`Timeout.timeout` throws `Timeout::ExitException`:
```
irb(main):001:0> require 'timeout'
=> true
irb(main):002:0> Timeout::ExitException.ancestors
=> [Timeout::ExitException, Exception, Object, Kernel, BasicObject]
```

This needs a test, of course.